### PR TITLE
add network.json validation

### PR DESCRIPTION
**Pull request by MaximilianMeister**
this will replace the check in the installation script
As long as the network.json is not valid, the user is not
allowed to start the installation

cc @vuntz

### CODE REVIEW

**Comment by MaximilianMeister** [comment]
not sure yet where or how to cache the last modification time of `network.json`, right now the validation script gets called on each request. any ideas?

**Comment by tboerger** [comment]
> not sure yet where or how to cache the last modification time of network.json, right now the validation script gets called on each request. any ideas?

Rails.cache?

File: crowbar_framework/lib/crowbar/installer.rb [hunk 1]
```diff
@@ -77,6 +78,10 @@ def initial_chef_client?
         steps_done.include?(:initial_chef_client)
       end
 
+      def valid_network_json?
+        Crowbar::Validator::NetworkJsonValidator.new.validate
```
Comment (tboerger) [inline crowbar_framework/lib/crowbar/installer.rb]: Please call it just `NetworkValidator`

**Comment by MaximilianMeister** [comment]
@tboerger @vuntz I have used `Rails.cache` now. please have another look

**Comment by MaximilianMeister** [comment]
I just realized that I need to adapt the `/installer/start` API in the back end as well, to not allow installation... will fix tomorrow.

**Comment by MaximilianMeister** [comment]
related PR for the crowbar-client is https://github.com/crowbar/crowbar-client/pull/43

**Comment by rsalevsky** [comment]
Can you drop the code from `crowbar_framework/lib/crowbar/checks/network.rb`? Otherwise we have some duplicated code.

**Comment by MaximilianMeister** [comment]
> Can you drop the code from crowbar_framework/lib/crowbar/checks/network.rb? Otherwise we have some duplicated code.

right, thx I forgot about it. it is not used by anything right now right?

**Comment by rsalevsky** [comment]
@MaximilianMeister Yes, it's not used at all.

**Comment by rsalevsky** [comment]
+1

**Comment by aplanas** [comment]
I do not like that `bin/network-json-validator` is called to validate the JSON document. IMHO this logic needs to be in a library used both in the CLI and here.

But I understand that this is not in the scope of the PR.

+1

**Pull request merged**